src/Integration.Vsix/Commands/ProjectSonarLintMenuCommand.cs

```diff
@@ -0,0 +1,53 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ProjectSonarLintMenuCommand.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using EnvDTE;
+using Microsoft.VisualStudio.Shell;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class ProjectSonarLintMenuCommand : VsCommandBase
+    {
+        private readonly IProjectPropertyManager propertyManager;
+
+        public ProjectSonarLintMenuCommand(IServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+            this.propertyManager = this.ServiceProvider.GetMefService<IProjectPropertyManager>();
+            Debug.Assert(this.propertyManager != null, $"Failed to get {nameof(IProjectPropertyManager)}");
+        }
+
+        protected override void InvokeInternal()
+        {
+            // Do nothing; this is a menu only.
+        }
+
+        protected override void QueryStatusInternal(OleMenuCommand command)
+        {
+            command.Enabled = false;
+            command.Visible = false;
+            if (this.propertyManager == null)
+            {
+                return;
+            }
+
+            IList<Project> projects = this.propertyManager
+                                          .GetSelectedProjects()
+                                          .ToList();
+
+            if (projects.Any() && projects.All(x => Language.ForProject(x).IsSupported))
```

Comment:
Because I also need to check if the user also has an unsupported project selected, together with a supported one.
E.g., selected one C# project and one C++ project. In this case there is still one supported project, but because there's also an unsupported project selected, the commands should be disabled & hidden.